compiler/rustc_mir/src/transform/coverage/spans.rs

```diff
@@ -344,18 +393,39 @@ impl<'a, 'tcx> CoverageSpans<'a, 'tcx> {
             } else if self.prev_original_span == self.curr().span {
                 // Note that this compares the new span to `prev_original_span`, which may not
                 // be the full `prev.span` (if merged during the previous iteration).
-                self.hold_pending_dups_unless_dominated();
+                if self.prev().is_macro_expansion() && self.curr().is_macro_expansion() {
+                    // Macros that expand to include branching (such as
```

Comment:
Idea for the future: I guess we don't have a way of also recording the coverage _inside_ the macro.

Reply:
Yes and no. If the macro defines the start and end (curly braces, usually, except in some closures) of a function body, then the coverage will be counted against the macro itself:

```
  670|       |            __fn_bitflags! {
  671|       |                /// Returns `true` all of the flags in `other` are contained within `self`.
  672|       |                #[inline]
  673|     94|                pub const fn contains(&self, other: $BitFlags) -> bool {
  674|     94|                    (self.bits & other.bits) == other.bits
  675|     94|                }
  ------------------
  | <clap::args::settings::Flags>::contains:
  |  673|     60|                pub const fn contains(&self, other: $BitFlags) -> bool {
  |  674|     60|                    (self.bits & other.bits) == other.bits
  |  675|     60|                }
  ------------------
  | <clap::app::settings::Flags>::contains:
  |  673|     34|                pub const fn contains(&self, other: $BitFlags) -> bool {
  |  674|     34|                    (self.bits & other.bits) == other.bits
  |  675|     34|                }
  ------------------
  676|       |            }
```

But the only reason that actual called functions get coverage (and called macros don't) is, of course, because the function has its own MIR.

It might be possible to add coverage spans to the macro spans, but we would have to execute the `mir_to_initial_sorted_coverage_spans()` logic multiple times, for each macro context (each macro invocation from the MIR source function, and nested macros). It's not inconceivable, but may be impractical.

Reply:
I added an example in the test of coverage of a function defined within a macro